aten/src/ATen/native/cpu/Loops.h

```diff
@@ -126,6 +160,22 @@ basic_loop(char* C10_RESTRICT data[], const int64_t* strides_, int64_t i, int64_
   execute_op(data, strides, i, n, op);
 }
 
+template <typename func_t>
+static inline void
+basic_loop(char* C10_RESTRICT data[], const int64_t* strides_, const ScalarType *dtypes, int64_t i, int64_t n, func_t op) {
+  using traits = function_traits<func_t>;
+  constexpr int ntensors = traits::arity + 1;
+
+  // Copying strides to temporary array helps auto vectorization in older GCC
+  // versions.
```

Comment:
which gcc versions need this? Note that gcc 5 is no longer supported, so workarounds for it are not necessary.

Reply:
I don't know. It was copy-pasted from the existing code and modified. Let me try to find the answer.

Reply:
I searched gcc's changelogs at for example https://gcc.gnu.org/gcc-7/changes.html for `vectoriz` for different gcc versions. The only thing that interests me is at https://gcc.gnu.org/gcc-5/changes.html. I don't know if this is related.

@colesbury might know the answer.